src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ContextMenuStripActionListTests.cs

```diff
@@ -0,0 +1,157 @@
+    private readonly Mock<IComponentChangeService> _mockComponentChangeService;
+    private readonly ToolStripDropDownMenu _toolStripDropDownMenu;
+    private readonly ToolStripDropDown _toolStripDropDown;
+    private readonly ToolStripDropDownDesigner _designer;
+    private readonly ContextMenuStripActionList _actionList;
+
+    public ContextMenuStripActionListTests()
+    {
+        _mockDesignerHost = new();
+        _mockSelectionService = new();
+        _mockComponentChangeService = new();
+        _toolStripDropDownMenu = new();
+        _toolStripDropDown = new();
+        InitializeMocks();
+
+        _designer = new();
+        _designer.Initialize(_toolStripDropDownMenu);
+        _actionList = new(_designer);
+
+        void InitializeMocks()
+        {
+            Mock<ISite> mockSite = new();
+            mockSite.Setup(s => s.GetService(typeof(IDesignerHost))).Returns(_mockDesignerHost.Object);
+            mockSite.Setup(s => s.GetService(typeof(ISelectionService))).Returns(_mockSelectionService.Object);
+            mockSite.Setup(s => s.GetService(typeof(IComponentChangeService))).Returns(_mockComponentChangeService.Object);
+            _toolStripDropDownMenu.Site = mockSite.Object;
+
+            _mockDesignerHost.Setup(d => d.GetService(typeof(IComponentChangeService))).Returns(_mockComponentChangeService.Object);
+        }
+    }
+
+    public void Dispose()
+    {
+        _toolStripDropDownMenu.Dispose();
+        _toolStripDropDown.Dispose();
+        _designer.Dispose();
+    }
+
+    [Fact]
+    public void Constructor_InitializesFields()
+    {
+        _actionList.Should().NotBeNull();
+        _actionList.Should().BeOfType<ContextMenuStripActionList>();
+
+        ToolStripDropDown toolStripDropDownValue = (ToolStripDropDown)_actionList.TestAccessor().Dynamic._toolStripDropDown;
+        ((ToolStripDropDownMenu)toolStripDropDownValue).Should().Be(_toolStripDropDownMenu);
+    }
+
+    [Theory]
+    [BoolData]
+    public void AutoShow_GetSet_ReturnsExpected(bool value)
+    {
+        _actionList.AutoShow.Should().BeFalse();
+
+        _actionList.AutoShow = value;
+        _actionList.AutoShow.Should().Be(value);
+    }
+
+    [Theory]
+    [BoolData]
+    public void ShowImageMargin_GetSet_ReturnsExpected(bool value)
+    {
+        _actionList.ShowImageMargin.Should().BeTrue();
+
+        _actionList.ShowImageMargin = value;
+        _actionList.ShowImageMargin.Should().Be(value);
+    }
+
+    [Theory]
+    [BoolData]
+    public void ShowCheckMargin_GetSet_ReturnsExpected(bool value)
+    {
+        _actionList.ShowCheckMargin.Should().BeFalse();
+
+        _actionList.ShowCheckMargin = value;
+        _actionList.ShowCheckMargin.Should().Be(value);
+    }
+
+    [Theory]
+    [InlineData(ToolStripRenderMode.System)]
+    [InlineData(ToolStripRenderMode.Professional)]
+    [InlineData(ToolStripRenderMode.ManagerRenderMode)]
+    [InlineData(ToolStripRenderMode.Custom)]
```

Comment:
If this is all of the values for `ToolStripRenderMode` consider changing this to 
```suggestion
    [EnumData<ToolStripRenderMode>]
```